tensorflow/contrib/verbs/rdma.cc

```diff
@@ -271,6 +271,10 @@ RdmaChannel::RdmaChannel(const RdmaAdapter* adapter, const string local_name,
     self_.lid = attr.lid;
     self_.qpn = qp_->qp_num;
     self_.psn = static_cast<uint32_t>(random::New64()) & 0xffffff;
+    union ibv_gid gid;
+    CHECK(!ibv_query_gid(adapter_->context_, (uint8_t) 1, 0, &gid)) << "Query gid";
```

Comment:
I think we must have from the user the port number.
There are a lot of rdma devices with 2 ports and we can't assume the first is the one configured.
We can (optionally) get the port number from the user in server creation and expose it here.
@junshi15 - we need it also for the device selection (today it take ibv_devices_list[0]). 
I can do that tomorrow with pleasure, unless you already have something.